pkg/kubelet/nodeconfig/log.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeconfig
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+)
+
+// This file contains shims for inserting nodeconfigLogFmt, but still reporting the call site of the message.
+
+const nodeconfigLogFmt = "nodeconfig controller: %s"
+
+func fatalf(format string, args ...interface{}) {
+	var s string
+	if len(args) > 0 {
+		s = fmt.Sprintf(format, args...)
+	} else {
+		s = format
+	}
+	glog.FatalDepth(1, fmt.Sprintf(nodeconfigLogFmt, s))
```

Comment:
In the interest of immediately stopping the controller when things are really really wrong without inserting a bunch of error-handling boilerplate, how do you feel about panics in the very low-level functions that are recovered in `Run()` and in the sync-loop? These are testable, unlike the fatal errors, allow us to return an error from `Run()`, and only require a very small change (`fatalf` calls `glog.ErrorDepth` and panics instead of calling `glog.FatalDepth`).

There is precedent in the Go standard library for using `panic` like this, see https://golang.org/src/encoding/json/decode.go (`func (d *decodeState) error(err error)`, which calls `panic`, and the `recover` call in `unmarshal`, for example).